Lib/sam.h

```diff
@@ -41,9 +48,1141 @@
 
 // ----- Macros -----
 
+#define SOFTWARE_RESET NOP //TODO: Lookup macro
+#define __disable_irq() asm volatile("CPSID i":::"memory");
+#define __enable_irq()  asm volatile("CPSIE i":::"memory");
+
+
 // ----- Variables -----
 
+/* IO definitions (access restrictions to peripheral registers) */
+/**
+    \defgroup CMSIS_glob_defs CMSIS Global Defines
+    <strong>IO Type Qualifiers</strong> are used
+    \li to specify the access to peripheral variables.
+    \li for automatic generation of peripheral register debug information.
+*/
+#ifdef __cplusplus
+  #define   __I     volatile             /*!< Defines 'read only' permissions                 */
+#else
+  #define   __I     volatile const       /*!< Defines 'read only' permissions                 */
+#endif
+#define     __O     volatile             /*!< Defines 'write only' permissions                */
+#define     __IO    volatile             /*!< Defines 'read / write' permissions              */
+
+
+/* ************************************************************************** */
+/*   PERIPHERAL ID DEFINITIONS FOR SAM4SD32C */
```

Comment:
We'll probably want to ifdef the sections that don't map to the other sam4s chips.
My guess is that they map very closely (if they aren't the same).